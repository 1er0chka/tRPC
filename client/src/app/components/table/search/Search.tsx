import React, { FunctionComponent, MouseEventHandler } from "react";
import styles from "./Search.module.scss";
import Image from "next/image";

interface ISearchProps {
  onClick: MouseEventHandler<HTMLDivElement>;
  searchInfo: string;
  setSearchInfo: React.Dispatch<React.SetStateAction<string>>;
}

const Search: FunctionComponent<ISearchProps> = ({
  onClick,
  searchInfo,
  setSearchInfo,
}) => {
  return (
    <div className={styles.input}>
      <input
        type={"text"}
        placeholder={"Search..."}
        value={searchInfo}
        onChange={(event) => setSearchInfo(event.target.value)}
      />
      <div className={styles.searchIcon}>
        <Image
          width={100}
          height={100}
          layout={"responsive"}
          onClick={onClick}
          src={"/resources/images/search.png"}
          alt={""}
        />
      </div>
    </div>
  );
};

export default Search;
