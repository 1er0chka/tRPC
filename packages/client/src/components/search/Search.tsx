import React, {FunctionComponent, MouseEventHandler} from 'react';
import styles from './Search.module.scss'

interface ISearchProps {
    onClick: MouseEventHandler<HTMLDivElement>;
    searchInfo: string;
    setSearchInfo: React.Dispatch<React.SetStateAction<string>>;
}

const Search: FunctionComponent<ISearchProps> = ({onClick, searchInfo, setSearchInfo}) => {
    return <div className={styles.input} data-testid="search">
        <input data-testid="search-input"
            type={"text"}
            placeholder={"Search..."}
            value={searchInfo}
            onChange={(event) => setSearchInfo(event.target.value)}
        />
        <img className={styles.searchIcon} data-testid="search-button" onClick={onClick} src={"/resources/images/search.png"} alt={""}/>
    </div>
};

export default Search;