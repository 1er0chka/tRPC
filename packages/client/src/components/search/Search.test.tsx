import {render, screen} from "@testing-library/react";
import Search from "./Search";
import userEvent from "@testing-library/user-event";

const setSearchInfo = jest.fn()
const onClick = jest.fn()

describe('Search Tests', () => {
    test('Snapshot test', () => {
        render(
            <Search onClick={onClick} searchInfo={''} setSearchInfo={setSearchInfo}/>
        )
        const search = screen.queryByTestId('search')
        expect(search).toMatchSnapshot()
    })
    test('Action test', async () => {
        render(
            <Search onClick={onClick} searchInfo={'coin'} setSearchInfo={setSearchInfo}/>
        )
        const button = screen.queryByTestId('search-button')
        if (button) {
            await userEvent.click(button)
            expect(onClick).toHaveBeenCalled()
        }
    })
})