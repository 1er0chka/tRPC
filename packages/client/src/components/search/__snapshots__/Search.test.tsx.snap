// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Tests Snapshot test 1`] = `
<div
  class="input"
  data-testid="search"
>
  <input
    data-testid="search-input"
    placeholder="Search..."
    type="text"
    value=""
  />
  <img
    alt=""
    class="searchIcon"
    data-testid="search-button"
    src="/resources/images/search.png"
  />
</div>
`;
